PowerValidatedSolutions.psm1

```diff
@@ -19611,6 +19611,78 @@ Function Get-NsxtLdap {
 }
 Export-ModuleMember -Function Get-NsxtLdap
 
+Function New-NsxtLdap {
+    <#
+        .SYNOPSIS
+        Add an LDAP identity source
+
+        .DESCRIPTION
+        The New-NsxtLdap cmdlet adds an LDAP identity source
+
+        .EXAMPLE
+        New-NsxtLdap -dcMachineName dc-sfo01 -protocol LDAPS -startTtls false -domain sfo.rainpole.io -baseDn "ou=Security Users,dc=sfo,dc=rainpole,dc=io" -bindUser svc-vsphere-ad@sfo.rainpole.io -bindPassword VMw@re1! -certificate Root64.cer
+        This example create an Active Directory Identity Source over LDAPS
+
+        .EXAMPLE
+        New-NsxtLdap -dcMachineName dc-sfo01 -protocol LDAP -startTtls false -domain sfo.rainpole.io -baseDn "ou=Security Users,dc=sfo,dc=rainpole,dc=io" -bindUser svc-vsphere-ad@sfo.rainpole.io -bindPassword VMw@re1!
+        This example create an Active Directory Identity Source over LDAPS
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$dcMachineName,
+        [Parameter (Mandatory = $true)] [ValidateSet("LDAP", "LDAPS")] [String]$protocol,
+        [Parameter (Mandatory = $true)] [ValidateSet('true','false')] [String]$startTtls,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$domain,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$baseDn,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$bindUser,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$bindPassword,
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$certificate
+    )
+
+    if (!$PsBoundParameters.ContainsKey("certificate") -and ($protocol -eq "LDAPS")) {
+        $certificate = Get-ExternalFileName -title "Select the Root CA Certificate File (.cer)" -fileType "cer" -location "default"
+    } elseif ($protocol -eq "LDAPS") {
+        if (!(Test-Path -Path $certificate)) {
+            Write-Error  "Certificate (cer) for Root Certificate Authority '$certificate' File Not Found"
+            Break
+        }
+    }
+
+    if ($protocol -eq "LDAPS") {
```

Comment:
Should it use the defaults unless there is a port override or do we just want to assume the defaults?